collector/collector.go

```diff
@@ -72,19 +66,48 @@ func (d *typedDesc) mustNewConstMetric(value float64, labels ...string) promethe
 	return prometheus.MustNewConstMetric(d.desc, d.valueType, value, labels...)
 }
 
-func NewExporter(address string, logger log.Logger) Exporter {
+type ExporterOption func(*Exporter)
```

Comment:
I would prefer a config struct style of doing things. This is similar to how we're implementing this in other exporters.

```go
type ChronyCollectorConfig struct {
	CollectSource   bool
	CollectTracking bool
	...
}
```

Assemble the struct var in `main` and pass it to the `New` func.

Reply:
Ok, I will follow the style as the consul_exporter does.